model-optimizer/extensions/middle/EmbeddingBagResolver.py

```diff
@@ -13,99 +13,57 @@
+                                               'mode is supported for node {}.'.format(node.id)
+            node_name = node.name
+            rename_node(node, node_name + '/TBR')
+            is_packed = False
+            if len(node.in_ports()) < 3 or node.in_port(2).disconnected():
+                is_packed = True
+                embedding_bag = EmbeddingBagPackedSum(graph, {'name': node_name}).create_node()
             else:
-                assert node.mode == 0
-
-                dense_shape = int64_array([offsets_shape[0], indices_shape[0]])
-                default_index = int64_array(weights_shape[0])
-                sweightedsum = create_op_with_const_inputs(graph, ExperimentalSparseWeightedSum,
-                                                           {2: dense_shape, 4: default_index},
-                                                           {'name': node.name + '/WeightedSum'})
-                if index_shape is None:
-                    index_shape = indices_shape[-1]
+                embedding_bag = EmbeddingBagOffsetsSum(graph, {'name': node_name}).create_node()
+                node.in_port(2).get_connection().set_destination(embedding_bag.in_port(2))
+            rename_node(embedding_bag, node_name)
+            node.in_port(0).get_connection().set_destination(embedding_bag.in_port(0))
+            node.in_port(1).get_connection().set_destination(embedding_bag.in_port(1))
+            node.out_port(0).get_connection().set_source(embedding_bag.out_port(0))
+            if len(node.in_ports()) == 4 and not node.in_port(3).disconnected():
+                if is_packed:
+                    node.in_port(3).get_connection().set_destination(embedding_bag.in_port(2))
                 else:
-                    merge_offsets = merge_offsets and index_shape == indices_shape[-1]
-                weighted_sum_nodes.append((sweightedsum, indices_shape[-1]))
+                    # connect per_sample_weights
+                    node.in_port(3).get_connection().set_destination(embedding_bag.in_port(4))
 
-                default_embeddings = np.zeros([1, weights_shape[-1]])
-                weights_concat = create_op_with_const_inputs(graph, Concat, {1: default_embeddings},
-                                                             {'axis': 0, 'in_ports_count': 2})
-                node.in_port(0).get_connection().set_destination(weights_concat.in_port(0))
-                node.in_port(1).get_connection().set_destination(sweightedsum.in_port(1))
-                weights_concat.out_port(0).connect(sweightedsum.in_port(3))
+                    weights_shape = embedding_bag.in_port(0).data.get_shape()
 
-                node.out_port(0).get_connection().set_source(sweightedsum.out_port(0))
-        self.create_offsets_for_weighted_sum(graph, weighted_sum_nodes, merge_offsets, index_shape)
+                    # expand embedding table with zeros
+                    default_embeddings = np.zeros([1, weights_shape[-1]])
```

Comment:
Line 61 can be implemented in a reshape-able way. We can take the Shape(embedding_bag.in_port(0)) then get the last dimension using `node_to_get_shape_value_of_indices`. Then Unsqueeze it. Then Broadcast Constant [0] to the calculated shape.

@jane-intel, FYI.